src/client.ts

```diff
@@ -499,8 +516,14 @@ export class Client {
       }
       this.logger.debug(`on deliverV1 -> ${consumer.consumerRef}`)
       this.logger.debug(`response.messages.length: ${response.messages.length}`)
-      await this.askForCredit({ credit: 1, subscriptionId: response.subscriptionId }, connection)
-      response.messages.map((x) => consumer.handle(x))
+      const creditRequestWrapper = this.askForCredit(response.subscriptionId, connection)
```

Comment:
Since this part is identical for v1 and v2 (maybe I'm wrong), should we extract it to a separate method taking the response, connection and consumer?

Reply:
We can refactor now or follow the rule of three, it's the same to me

Reply:
Touche'